### PR TITLE
Toast surface + workspace UX fixes

### PR DESCRIPTION
**Pull request by maui1911**
## Summary

A bundle of UX fixes around the workspace + a full redesign of the notification surface.

### Toast surface (popup-hosted)
- New `IToastService` + popup-hosted `ToastHostView` that defeats the WPF airspace bug — toasts now render above the terminal HwndHost children. Previously errors fell *behind* terminals.
- Visual spec follows `docs/design/html/CodeScope - Errors and Toasts.html` exactly: 380×auto, severity rail, 24×24 icon tile, 2px meter, 28px anchor.
- Severity-aware behavior: info/ok 4s, warn 8s, **err persistent**. Hover anywhere on the stack pauses every meter; resume preserves remaining time. MaxVisible=3 cap on non-errors only — errors never auto-fold.
- `ErrToast` helper gives error sites Copy + optional Retry actions. Retry is awaited inside try/catch with a follow-up err toast on failure (no silent task faults).
- New `Fig.Style.Button.ToastAction` chip-style button in `DesignTokens.xaml`, severity-aware primary swap.

### Workspace + sidebar
- **Click on terminal area inside a split group focuses that group** (HwndHost was eating mouse events; backstopped via `GotKeyboardFocus` on the EditorGroupView).
- **Sidebar's blue rail follows the focused tab** across split-groups. `MainViewModel.SyncSidebarToTab` resolves the tab → owning project → owning worktree (project-scoped lookup; worktree ids are not globally unique). `SidebarView` now drives the visual TreeView selection from the VM via `WorktreeSelected` + `ItemContainerGenerator`.
- **Auto-prune worktrees whose folder was deleted outside the app**. `WorktreeStatusPoller` counts ~6s of consecutive misses, then prunes; emits a Warn toast (no more silent removal). `SessionStore.PruneMissingWorktreeAsync` does mutate-then-persist with a *narrow* rollback that doesn't clobber concurrent worktree mutations on save failure.

### Terminal paste
- Bracketed paste mode (`ESC[200~…ESC[201~`) instead of CRLF normalization — multi-line pastes now land as one paste event in Claude Code/Codex/modern shells. Raw cmd.exe loses paste support but that was an explicit trade-off ("fuck raw cmd.exe").
- `Ctrl+V` `e.Handled = true` set before clipboard read so the native paste path can't race.

### Infra
- `AutomationProperties.AutomationId` on dynamic tree/list items, status-bar widgets, toast root + actions + close button. Wpf-cli refs resolve in O(depth) instead of full-tree walks.
- `WorktreeViewModel` raises `OnPropertyChanged(nameof(AutomationId))` on branch rename so refs don't go stale.
- `ConfirmDialog.Show` no longer assigns Owner=self when called before MainWindow exists (caused a startup crash on a single-instance race).
- Cached + Frozen severity-icon `Geometry` instances.
- `.scratch/` added to `.gitignore`.

## Codex review

Three rounds of multi-agent code review (correctness/bugs, WPF idioms, UX/spec fidelity) plus a fix-verification round. Findings addressed inline; final round confirmed ready to ship. Only minor non-blocking notes remain (SidebarView event subscription doesn't unsubscribe on Unloaded — but VM is a DI singleton, view is permanent, no real leak).

## Test plan

- [x] Build clean: 0 errors
- [x] Tests: 150/151 pass (1 pre-existing FSWatcher flake, unrelated)
- [x] Smoke tested via wpf-cli: tabs in two different worktrees, clicking between them moves both the editor-group rail and the sidebar selection in both directions
- [x] Toast sample (`CODESCOPE_TOAST_SAMPLE=1`) renders all four severities to spec, popup overlays the terminal correctly
- [ ] Manual exercise of Retry actions (Fetch/Pull/CreatePR) on real network failures
- [ ] Manual exercise of worktree auto-prune (delete folder outside app, watch for warn toast)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/CodeScope.Ui/ViewModels/SessionTabViewModel.cs [hunk 1]
```diff
@@ -83,6 +83,23 @@ public void Rebind(SessionDescriptor descriptor)
...
+    /// snapshots will show e.g. <c>a:Tab_main</c> instead of the VM type name.
+    /// </summary>
+    public string AutomationId
+        => $"Tab_{(string.IsNullOrWhiteSpace(DisplayName) ? Descriptor.Id : SafeToken(DisplayName))}";
+
+    private static string SafeToken(string s)
+        => string.IsNullOrWhiteSpace(s)
+            ? "unknown"
+            : new string([.. s.Select(c => char.IsLetterOrDigit(c) ? c : '_')]).Trim('_');
+
```
[the remark this answers is not in the document]
Comment (maui1911) [reply src/CodeScope.Ui/ViewModels/SessionTabViewModel.cs]: Fixed in cbed3ef: SafeToken now returns "unknown" when the trimmed result is empty (all-punctuation inputs). Applied consistently across ProjectViewModel, WorktreeViewModel, and SessionTabViewModel.

File: src/CodeScope.App/Toasts/ToastItemViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+            (request.Actions ?? []).Select(a => new ToastActionViewModel(a, this)));
+        Progress = 1.0;
+
+        // Persistent toasts (errors by default) draw no meter and don't tick.
+        if (duration == Timeout.InfiniteTimeSpan || duration == TimeSpan.Zero)
+        {
+            HasMeter = false;
+            Progress = 1.0;
+            return;
+        }
```
[the remark this answers is not in the document]
Comment (maui1911) [reply src/CodeScope.App/Toasts/ToastItemViewModel.cs:55]: Fixed in cbed3ef: TimeSpan.Zero now falls through to the ticker path and auto-dismisses on the first 60ms tick. Only Timeout.InfiniteTimeSpan is the persistent sentinel.

File: src/CodeScope.Ui/Views/SidebarView.xaml.cs [hunk 3]
```diff
@@ -11,9 +11,65 @@ namespace NoScope.CodeScope.Ui.Views;
...
+        // changes never moved the blue rail.
+        if (_wiredVm is not null)
+        {
+            _wiredVm.WorktreeSelected -= OnWorktreeSelectedFromVm;
+        }
+        _wiredVm = e.NewValue as SidebarViewModel;
+        if (_wiredVm is not null)
+        {
+            _wiredVm.WorktreeSelected += OnWorktreeSelectedFromVm;
+        }
```
[the remark this answers is not in the document]
Comment (maui1911) [reply src/CodeScope.Ui/Views/SidebarView.xaml.cs:45]: Fixed in cbed3ef: added Unloaded handler that unsubscribes from WorktreeSelected and nulls _wiredVm.

File: src/CodeScope.App/Toasts/ToastHostView.xaml.cs [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+    /// <summary>
+    /// Position the popup so its bottom-right corner sits at the window's bottom-right
+    /// minus a 28px gutter. <paramref name="targetSize"/> is the parent window's
+    /// render size; <paramref name="popupSize"/> is the measured popup contents.
+    /// </summary>
+    private static CustomPopupPlacement[] PlaceBottomRight(Size popupSize, Size targetSize, Point offset)
+    {
+        var x = targetSize.Width - popupSize.Width - Inset;
+        var y = targetSize.Height - popupSize.Height - Inset;
```
[the remark this answers is not in the document]
Comment (maui1911) [reply src/CodeScope.App/Toasts/ToastHostView.xaml.cs]: Fixed in cbed3ef: X and Y are now clamped to Math.Max(0, ...) so the toast stack stays on-screen when the window is smaller than the popup.

File: src/CodeScope.App/Toasts/ToastView.xaml [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+                        </ItemsControl.ItemsPanel>
+                        <ItemsControl.ItemTemplate>
+                            <DataTemplate>
+                                <Button Command="{Binding InvokeCommand}"
+                                        Style="{StaticResource Fig.Style.Button.ToastAction}"
+                                        Margin="0,0,8,0"
+                                        AutomationProperties.AutomationId="{Binding Label, StringFormat='ToastAction_{0}'}"
+                                        AutomationProperties.Name="{Binding Label}">
+                                    <TextBlock Text="{Binding Label}" />
+                                </Button>
```
[the remark this answers is not in the document]
Comment (maui1911) [reply src/CodeScope.App/Toasts/ToastView.xaml:110]: Fixed in cbed3ef: toast action AutomationId is now generated from Sanitize(Label) + owning toast id (e.g. ToastAction_Retry_abc123), exposed as a property on ToastActionViewModel. No more raw label text in UIA ids.

File: src/CodeScope.Ui/ViewModels/SessionTabViewModel.cs [hunk 6]
```diff
@@ -83,6 +83,23 @@ public void Rebind(SessionDescriptor descriptor)
...
     public string Title => DisplayName;
 
-    partial void OnDisplayNameChanged(string value) => OnPropertyChanged(nameof(Title));
+    /// <summary>
+    /// Stable, human-readable id for UIA automation. Falls back to the descriptor's
+    /// session id when DisplayName is empty so a:Tab_* always resolves. Wpf-cli
+    /// snapshots will show e.g. <c>a:Tab_main</c> instead of the VM type name.
+    /// </summary>
+    public string AutomationId
+        => $"Tab_{(string.IsNullOrWhiteSpace(DisplayName) ? Descriptor.Id : SafeToken(DisplayName))}";
```
[the remark this answers is not in the document]
Comment (maui1911) [reply src/CodeScope.Ui/ViewModels/SessionTabViewModel.cs]: Fixed in cbed3ef: the AutomationId fallback now applies SafeToken to Descriptor.Id as well, so Tab_* ids are uniformly sanitized. (In practice Descriptor.Id is a Guid hex string already safe, but consistency across the codebase is better.)

File: src/CodeScope.Core/Services/SessionStore.cs [hunk 7]
```diff
@@ -400,6 +400,73 @@ public async Task<Result<bool>> RemoveWorktreeAsync(string projectId, string wor
...
+                        rollbackSessions.Where(s => !existingSessionIds.Contains(s.Id))).ToList();
+                    _projects[idx] = p with { Worktrees = wts, Sessions = sessions };
+                }
+            }
+            return saved;
+        }
+
+        Raise(new SessionStoreChange.WorktreeRemoved(projectId, worktreeId));
+        return Result<bool>.Ok(true);
+    }
```
[the remark this answers is not in the document]
Comment (maui1911) [reply src/CodeScope.Core/Services/SessionStore.cs:468]: Agreed — PruneMissingWorktreeAsync has enough edge-case logic (narrow rollback, primary guard, session cascade) that test coverage would be valuable. Tracking this as a follow-up for the next session rather than expanding this PR further.

File: src/CodeScope.App/Toasts/ToastItemViewModel.cs [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+    public string Label => _model.Label;
+    public bool IsPrimary => _model.IsPrimary;
+    /// <summary>Forwarded so the action button's accent can follow the toast severity.</summary>
+    public ToastSeverity Severity => _owner.Severity;
+
+    [RelayCommand]
+    private void Invoke()
+    {
+        try { _model.Invoke(); }
+        finally { _owner.DismissCommand.Execute(null); }
```
[the remark this answers is not in the document]
Comment (maui1911) [reply src/CodeScope.App/Toasts/ToastItemViewModel.cs:162]: This is by design — the Retry action is a synchronous `Action` that fires `_ = RunRetryAsync(retry, label)` (see `SidebarViewModel.ErrToast`). The async retry runs in the background; the toast dismisses immediately because the user already chose to act. If the retry fails, `RunRetryAsync` catches and surfaces a *new* error toast with the failure message. Keeping the original toast open during the async would leave a stale "Fetch failed" toast visible while either (a) the retry succeeds (confusing) or (b) the retry shows a fresh error (duplicate).

File: src/CodeScope.App/Toasts/ToastService.cs [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+        // If everything visible is an error and the user keeps stacking errors, we let
+        // the stack grow past MaxVisible so the user sees them all rather than silently
+        // dropping the oldest critical message on the floor.
+        while (Items.Count(i => i.Severity != ToastSeverity.Err) > MaxVisible)
+        {
+            var victim = Items.FirstOrDefault(i => i.Severity != ToastSeverity.Err);
+            if (victim is null) { break; }
+            victim.StopTimer();
+            Items.Remove(victim);
+        }
```
[the remark this answers is not in the document]
Comment (maui1911) [reply src/CodeScope.App/Toasts/ToastService.cs:59]: Agreed this would improve the UX for burst scenarios. The spec mentions folding/overflow affordance but the current toast volume (sidebar operations, update checks) rarely exceeds 3 simultaneously. Tracking as a follow-up enhancement — adding a "+N more" collapsed footer and an overflow queue is a meaningful UI addition that deserves its own design pass rather than being shoehorned into this PR.

File: src/CodeScope.Ui/Services/IToastService.cs [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+    Warn = 2,
+    Err = 3,
+}
+
+/// <summary>
+/// One inline action button on a toast. Maximum two per toast (spec §02 callout 5).
+/// Click resolves before the toast auto-dismisses, so an async handler can finish its
+/// work before the row disappears.
+/// </summary>
+public sealed record ToastAction(string Label, Action Invoke, bool IsPrimary = false);
```
[the remark this answers is not in the document]
Comment (maui1911) [reply src/CodeScope.Ui/Services/IToastService.cs:21]: The async boundary is currently handled at the callsite level: ErrToast accepts Func<Task>? retry and wraps it in RunRetryAsync (which awaits, catches, and shows a follow-up toast on failure). The ToastAction.Invoke is deliberately synchronous because dismiss-immediately-after-click is the right UX for Retry — the user clicked it, the toast should go away, and the retry result appears as a new toast. Making Invoke async and awaiting before dismiss would leave a stale "Fetch failed" toast visible for 10+ seconds during a network retry. If we add non-retry async actions in future, we can introduce an AsyncToastAction variant.
